daishi4u/ForceCode
# Test github action (#124)

* 3.17.1 (#291)

* Update build related files (#290)

* Update build files

* Add .gitlab-ci.yml to .vscodeignore

* Remove git push --tags

* 3.17.1-0

* Update .gitlab-ci.yml

* Move vscode:prepublish commands to package command

* [ci skip] Update default API Version

* Fix users not receiving error message when SFDX wasn't installed on startup

* 3.17.1

* 3.17.2 (#296)

* Fix issue with logging into orgs via saved usernames view

* Clean old versions from the CI cache when building

* Fix issue with viewing failed deployment. Remove unneeded code

* Upgrade to newest vscode API

* Remove commandService

* Open Lightning when checking deployment status

* Open Lightning when opening file in org

* 3.17.2-2

* Fix creating a scratch org

* 3.17.2-3

* Update the README

* Begin work on unit tests

* Separate install, build, and package

* Azure test (#295)

* Begin test migration to azure

* Update coverage to be used with azure

* Add build and future release job

* Move strategy inside test job

* Add publish script

* Add extra tasks for build and deploy

* Fix build artifacts

* Try file copy as non-glob pattern

* Use Azure as primary CI/CD

* Download previous built artifact in pipeline

* Update pipline for download

* 3.17.2

* 3.17.3 (#298)

* Re-combine build and package

* Separate createProject command from extension

* Unit test work

* Fix issues retrieving debug logs

* Update changelog

* 3.17.3

* Update azure-pipelines.yml

* 3.17.4

Fix building the extension before publish

* 3.18.0 (#304)

* Add option to cancel Apex save when it times out

* Convert object into array<array<{}>> for ease of use

* Add time saved to end of SUCCESS! in save history view, optimization

* Add number of lines covered in Forcecode Code Coverage view hovers

* Clear tooltip when clearing coverage

* Add runTest command to code coverage view

* 3.17.5-2

* Fix error related to ForceCode.treeDataProvider

* Update changelog

* Query metadata files while checking types on start

* Don't read configuration twice on startup

* Remove unneeded code

* Show execute anon output in document. Implement logging

* Add warning to notifications

* Show debug logs with dated names

* 3.18.0-1

* Remove unneeded folder check

* Add missing period in extension tooling type check

* Add label in to covered class tooltip

* Fix showing sfdx errors

* Add inline cancel task

* Exclude package.xml from file chooser in Choose Files deploy option

* 3.18.0-2

* Fix get debug log

* Reload window when user selects open folder as new project path

* Fix various menus throwing errors when user didn't make a selection

* 3.18.0-3

* Update jsforce. Get rid of glob. Add bulkLoaderPollTimeout

* 3.18.0-4

* Don't ask for analytics during testing

* Don't ask for analytics part 2

* Integration test work (#303)

* Test cli login during testing

* Test running test logged in

* Create settings file for tests when logged in

* Use real SFDX commands if logged in for test

* Don't stub when logged in

* Add more tests

* Even more tests

* Add retrieve, save, diff, overall coverage to tests

* Fix execute anonymous when project folder contains a space

* Split test files

* Add find, getLog, and executeAnonymous to test suite

* Add apexTest to test suite

* Set active editor

* 3.18.0-5

* Allow editor to take focus in test

* Small test fixes

* Test fixup

* Move debug check to fauxClassGenerator

* Fix execute anonymous test

* Unit test createClass, executeAnonymous

* Use auto-compile for LWC test

* Verify code coverage view has contents in test

* 3.18.0

* 3.18.1 (#305)

* Disable extensions while testing

* Test save fail and save pass on each file type

* Add auto-compile test

* Fix not showing errors when metadata has a syntax error. Test saving metadata

* 3.18.1-1

* Add createScratchOrg test

* Add dxCommands.ts tests

* Add pipeline and coverage badges to readme

* Add more retrieve tests

* Add static resource zip auto-compile test

* Fix name in static resource test

* Static resource testing

* apexTest.ts testing

* Fix not refreshing an expired access token when saving a static resource

* 3.18.1-2

* Update CI build file

* Update coverconfig

* Add cancel command test

* Fix forcecodeCommand cancellation test

* Update build. Cancel retrieve instead of CCR

* Revert built update

* 3.18.1

* 3.18.2 (#311)

* Always feed notification logic a string

* Fix "Relationship fields not displayed #306"

* 3.18.2-1

* Fix output.getKeys

* 3.18.2

* 3.18.3 (#313)

* Fix retrieving unmanaged packages

* 3.18.3-1

* Place package metadata directly in src folder

Allow unmanaged package metadata to work with the Forcecode workflow

* Fix deploying certain metadata types e.g. custom metadata records

* 3.18.3
